search/saddleback_search.cpp

```diff
@@ -0,0 +1,106 @@
+/**
+ * @file
+ * @brief Implementation of [Saddleback Algorithm](https://www.geeksforgeeks.org/saddleback-search-algorithm-in-a-2d-array) for 2D arrays.
+ *
+ * @details
+ * Saddleback Algorithm is an algorithm that searches 2D array in linear time,
+ * i.e, O(m + n), where m is row and n is column of 2D array. Also, each row and
+ * column of the matrix should be sorted beforehand for this algorithm to work.
+ *
+ * @author [Hashir Niazi](https://github.com/HashirGJ8842)
+ */
+#include <cassert>      /// for assert
+#include <iostream>     /// for io operations, and std::pair
+#include <vector>       /// for std::vector
+
+/** \namespace search
+ * \brief Algorithms for searching
+ */
+namespace search {
+/** \namespace saddleback
+ * \brief Function for implementing [Saddleback Algorithm](https://www.geeksforgeeks.org/saddleback-search-algorithm-in-a-2d-array).
+ */
+namespace saddleback {
+/**
+ * This function implements [Saddleback Algorithm](https://www.geeksforgeeks.org/saddleback-search-algorithm-in-a-2d-array),
+ * on a sorted 2D array, and finds the location of the element needed to search
+ * @param matrix 2D matrix which is sorted on the basis of rows and columns
+ * @param element the element which is needed to be searched
+ * @return An std::pair of with row and column populated within it, if the
+ * element is present.
+ * @return An std::pair with (0, 0), if the element is not present.
+ */
+std::pair<uint32_t, uint32_t> saddleback(std::vector<std::vector<int32_t>> matrix,
+                               int32_t element) {
+    uint32_t left_index = 0;
+    uint32_t right_index = matrix[0].size() - 1;  // Start from top right corner
+    while (left_index < matrix.size()) {  // Exit once the value of indexes get out of range.
+        if (element ==
+            matrix[left_index]
+                  [right_index]) {  // If value on this position of matrix is
+                                    // equal to element, return (row, column).
+            return std::pair<uint32_t, uint32_t>(left_index+1, right_index+1);
```

Comment:
more appropriate way is
```suggestion
            return std::make_pair(left_index+1, right_index+1);
```